File: 백준/Bronze/25304. 영수증/영수증.java
import java.util.Scanner;
public class Main {

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int total = scan.nextInt();
		int how_many_total = scan.nextInt();
		int our_calculate = 0;
		
		while (how_many_total>0) {
			int price = scan.nextInt();
			int how_many = scan.nextInt();
			our_calculate += price*how_many;
			how_many_total--;
		}
		if (total == our_calculate) {
			System.out.println("Yes");
		} else {
			System.out.println("No");
		}
	}

}

File: 백준/Bronze/25304. 영수증/영수증.cc
#include <iostream>
using namespace std;

int main(void){
    int X, N, a, b;
    cin >> X >> N;
    for (int i=0; i<N; i++){
        cin >> a >> b;
        X -= a*b;
    }
    // 삼항연산자 사용
    // 맞으면 ? 뒤에 실행
    // 틀리면 : 뒤에 실행
    (X==0) ? cout << "Yes" : cout << "No";
    return 0;
}